controllers/apps/transformer_component_workload.go

```diff
@@ -680,41 +680,28 @@ func (r *componentWorkloadOps) leaveMember4ScaleIn(deleteReplicas, joinedReplica
 }
 
 func (r *componentWorkloadOps) leaveMemberForPod(pod *corev1.Pod, pods []*corev1.Pod) error {
-	isLeader := func(pod *corev1.Pod) bool {
-		if pod == nil || len(pod.Labels) == 0 {
-			return false
-		}
-		roleName, ok := pod.Labels[constant.RoleLabelKey]
-		if !ok {
-			return false
-		}
-
-		for _, replicaRole := range r.runningITS.Spec.Roles {
-			if roleName == replicaRole.Name && replicaRole.IsLeader {
-				return true
+	trySwitchover := func(lfa lifecycle.Lifecycle, pod *corev1.Pod) error {
+		err := lfa.Switchover(r.reqCtx.Ctx, r.cli, nil, "")
```

Comment:
It's better to pass the observed role to it.

Reply:
Observed role will be retrived when setting KB_SWITCHOVER_ROLE env.